app/reducers/companySemesters.js

```diff
@@ -12,26 +13,25 @@ export type CompanySemesterEntity = {
   activeInterestForm?: boolean
 };
 
+type State = any;
+
 export default createEntityReducer({
   key: 'companySemesters',
   types: {
     fetch: Company.FETCH_SEMESTERS
   },
-  mutate(state, action) {
-    switch (action.type) {
-      case Company.ADD_SEMESTER.SUCCESS: {
-        return {
-          ...state,
-          byId: {
-            ...state.byId,
-            [action.payload.id]: action.payload
-          },
-          items: state.items.concat(action.payload.id)
-        };
+  // TODO: I think this can be removed by using { types: mutate: Company.ADD_SEMESTER} above
```

Comment:
Mutate is for dealing with optimistic stuff. If the result is in the response, and the schema is set; createEntityReducer should do this by default i think.